dotCMS/src/main/java/com/liferay/util/FileUtil.java

```diff
@@ -256,6 +260,59 @@ public static void deltree(String directory) {
 		deltree(new File(directory));
 	}
 
+	/**
+   * This method takes a directory and will delete all files older than
+   * the passed in Date.  It will only delete directories older than the
+   * passed in Date if they are empty. It will not delete the directory passed in
+	 * @param directory
+	 * @param olderThan
+	 */
+  public static void cleanTree(File directory, Date olderThan) {
+    cleanTree(directory, olderThan.getTime()); 
+  }
+  
+  /**
+   * This method takes a directory and will delete all files older than
+   * the passed in time in millis.  It will only delete directories older than the
+   * passed in time if they are empty.  It will not delete the directory passed in
+   * @param directory
+   * @param deleteOlderTime
+   */
+	 public static void cleanTree(final File directory, long deleteOlderTime) {
+
+	    if (directory==null || !directory.exists() || !directory.isDirectory()) {
+	      Logger.info(FileUtil.class, "cleanTree Directory " + directory + " not found exiting");
+	      return;
+	    }
+
+	    final List<File> allFiles = com.liferay.util.FileUtil.listFilesRecursively(directory, new FileFilter() {

```

Comment:
I think you may replace this by lambda such as
```
pathname -> pathname.lastModified() < deleteOlderTime
```

Probably deleteOlderTime should be final